src/core/openthread.cpp

```diff
@@ -181,6 +181,11 @@ void otGetFactoryAssignedIeeeEui64(otInstance *aInstance, otExtAddress *aEui64)
     otPlatRadioGetIeeeEui64(aInstance, aEui64->m8);
 }
 
+void otGetHashMacAddress(otInstance *, otExtAddress *aHashMacAddress)
+{
+    sThreadNetif->GetMac().GetHashMacAddress(static_cast<Mac::ExtAddress *>(aHashMacAddress));
```

Comment:
I think it's ok to use `static_cast` here, please let me know if you have any concern？Thanks.